lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4SlaveFactory.scala

```diff
@@ -4,17 +4,33 @@ import spinal.core._
 import spinal.lib._
 import spinal.lib.bus.misc._
 
-object Axi4SlaveFactory {
-  def apply(bus: Axi4) = new Axi4SlaveFactory(bus)
-}
+case class Axi4SlaveFactory(bus: Axi4, cmdPipeline: StreamPipe = StreamPipe.M2S, addrRemapFunc: UInt => UInt = null) extends BusSlaveFactoryDelayed {
+  implicit class RichMultiData[T <: MultiData](b: T) {
+    def mapElement[E <: Data](locator: T => E)(f: E => E): T = {
+      val ret = b.clone.asInstanceOf[T].allowOverride()
+      ret := b
+      locator(ret) := f(locator(b))
```

Comment:
locator(ret).removeAssignments() := f(locator(b))

Would remove the need of allowOverride and avoid the double assignement.